src/vivarium_csu_ltbi/tools/builder.py

```diff
@@ -1,43 +1,57 @@
-cause_latent_tuberculosis_infection_954 = 954
-cause_drug_sus_tb_934 = 934
-cause_m_drug_res_tb_946 = 946
-cause_ext_d_res_tb_947 = 947
-hiv_d_sus_tb_948 = 948
-hiv_m_d_res_tb_949 = 949
-hiv_ext_d_res_tb_950 = 950
-hiv_all_other_300 = 300
-tb_297 = 297
-hiv_298 = 298
-
-#  Prevalence state names
-SUS_TB_SUS_HIV = 'sus_tb_sus_hiv'
-LTBI_SUS_HIV = 'ltbi_sus_hiv'
-ACTTB_SUS_HIV = 'acttb_sus_hiv'
-REC_LTBI_SUS_HIV = 'rec_ltbi_sus_hiv'
-REC_LTBI_PLUS_HIV = 'rec_ltbi_plus_hiv'
-SUS_TB_PLUS_HIV = 'sus_tb_plus_hiv'
-LTBI_PLUS_HIV = 'ltbi_plus_hiv'
-ACTTB_PLUS_HIV = 'acttb_plus_hiv'
+DEFAULT_PATH = gbd.ARTIFACT_FOLDER / PROJ_NAME
+
+#  state names
+ACTIVETB_POSITIVE_HIV = 'activetb_positive_hiv'
+ACTIVETB_SUSCEPTIBLE_HIV = 'activetb_susceptible_hiv'
+LTBI_SUSCEPTIBLE_HIV = 'ltbi_susceptible_hiv'
+LTBI_POSITIVE_HIV = 'ltbi_positive_hiv'
+RECOVERED_LTBI_SUSCEPTIBLE_HIV = 'recovered_ltbi_susceptible_hiv'
+RECOVERED_LTBI_POSITIVE_HIV = 'recovered_ltbi_positive_hiv'
+SUSCEPTIBLE_TB_POSITIVE_HIV = 'susceptible_tb_positive_hiv'
+SUSCEPTIBLE_TB_SUSCEPTIBLE_HIV = 'susceptible_tb_susceptible_hiv'
+
+# transition names (from, to)
+ACTIVETB_SUSCEPTIBLE_HIV_TO_ACTIVETB_POSITIVE_HIV = 'activetb_susceptible_hiv_to_activetb_positive_hiv'
+ACTIVETB_SUSCEPTIBLE_HIV_SUSCEPTIBLE_TB_SUSCEPTIBLE_HIV = 'activetb_susceptible_hiv_susceptible_tb_susceptible_hiv'
+ACTIVETB_POSITIVE_HIV_TO_SUSCEPTIBLE_TB_POSITIVE_HIV = 'activetb_positive_hiv_to_susceptible_tb_positive_hiv'
+LTBI_SUSCEPTIBLE_HIV_TO_RECOVERED_LTBI_SUSCEPTIBLE_HIV = 'ltbi_susceptible_hiv_to_recovered_ltbi_susceptible_hiv'
+LTBI_SUSCEPTIBLE_HIV_TO_ACTIVETB_SUSCEPTIBLE_HIV = 'ltbi_susceptible_hiv_to_activetb_susceptible_hiv'
+LTBI_SUSCEPTIBLE_HIV_TO_LTBI_POSITIVE_HIV = 'ltbi_susceptible_hiv_to_ltbi_positive_hiv'
+LTBI_POSITIVE_HIV_TO_RECOVERED_LTBI_POSITIVE_HIV = 'ltbi_positive_hiv_to_recovered_ltbi_positive_hiv'
+LTBI_POSITIVE_HIV_ACTIVETB_POSITIVE_HIV = 'ltbi_positive_hiv_activetb_positive_hiv'
+SUSCEPTIBLE_TB_POSITIVE_HIV_TO_LTBI_POSITIVE_HIV = 'susceptible_tb_positive_hiv_to_ltbi_positive_hiv'
+SUSCEPTIBLE_TB_SUSCEPTIBLE_HIV_TO_LTBI_SUSCEPTIBLE_HIV = 'susceptible_tb_susceptible_hiv_to_ltbi_susceptible_hiv'
+SUSCEPTIBLE_TB_SUSCEPTIBLE_HIV_TO_SUSCEPTIBLE_TB_POSITIVE_HIV = 'susceptible_tb_susceptible_hiv_to_susceptible_tb_positive_hiv'
+RECOVERED_LTBI_SUSCEPTIBLE_HIV_TO_SUSCEPTIBLE_TB_SUSCEPTIBLE_HIV = 'recovered_ltbi_susceptible_hiv_to_susceptible_tb_susceptible_hiv'
+RECOVERED_LTBI_SUSCEPTIBLE_HIV_TO_RECOVERED_LTBI_POSITIVE_HIV = 'recovered_ltbi_susceptible_hiv_to_recovered_ltbi_positive_hiv'
+RECOVERED_LTBI_POSITIVE_HIV_TO_SUSCEPTIBLE_TB_POSITIVE_HIV = 'recovered_ltbi_positive_hiv_to_susceptible_tb_positive_hiv'
 
 
 class DataRepo:
 
     def __init__(self):
-        pass
+        self._df_template = None
+        self.df_zero = None
+
+    def get_zeros(self):
+        return self.df_zero
+
+    def get_filled_with(self, fill_value):
+        return pd.DataFrame().reindex_like(self._df_template).fillna(fill_value)
```

Comment:
the only thing that weirds me out about this is it'll fail (I think) if you haven't set the template, and making the template is the same reindex-like call, too. Maybe you could just pass in the the template df with the fill value?

Reply:
Yes, it fails right away if the template is invalid. That seems OK because everything needs to succeed for the artifact to be valid.